### ISSUE TITLE
Base rule extension: no-undef should support the NodeJS namespace for types

### ISSUE THREAD

**Issue by PythonCoderAS** [typescript-eslint/typescript-eslint#5173]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+new+base+rule+extension%22) and found none that match my proposal.
- [X] I have searched the [current extension rule list](https://typescript-eslint.io/rules/#extension-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] My proposal specifically checks TypeScript syntax, or it proposes a check that requires type information to be accurate.
- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the base rule

https://eslint.org/docs/rules/no-undef

### Description

I am using attributes of the `NodeJS` namespace in order to type hint functions (`NodeJS.Signals` in particular). Eslint is saying that it cannot find the `NodeJS` variable, but it is only used for the type and so should not be triggered. In the resulting JS code type signatures are stripped out anyways.

### Fail

```typescript
var replace: string = doesNotExist;
```

### Pass

```typescript
const replace: NodeJS.Signals = 'SIGTERM';
```

**Comment by bradzacher** [comment]
Please read the FAQ as you ticked
https://typescript-eslint.io/docs/linting/troubleshooting#i-get-errors-from-the-no-undef-rule-about-global-variables-not-being-defined-even-though-there-are-no-typescript-errors
